diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/constants/Warnings.kt

```diff
@@ -103,17 +105,25 @@ enum class Warnings(private val canBeAutoCorrected: Boolean, private val warn: S
                    isFixMode: Boolean,
                    freeText: String,
                    offset: Int,
+                   node: ASTNode?,
                    canBeAutoCorrected: Boolean = this.canBeAutoCorrected,
                    autoFix: () -> Unit) {
-        warn(configRules, emit, canBeAutoCorrected, freeText, offset)
+        if (node != null && node.hasSuppress(name))
+            return
+        warn(configRules, emit, canBeAutoCorrected, freeText, offset, node)
         fix(configRules, autoFix, isFixMode)
     }
 
+    @Suppress("LongParameterList")
     fun warn(configs: List<RulesConfig>,
              emit: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit),
              autoCorrected: Boolean,
              freeText: String,
-             offset: Int) {
+             offset: Int,
+             node: ASTNode?) {
+        if (node != null && node.hasSuppress(name))
+            return
+
         if (configs.isRuleEnabled(this)) {
```

Comment:
I'd rather add `node.hasSupress(name)` in this condition, in both `warn` and `fix` methods.